src/lib.rs

```diff
@@ -73,8 +73,8 @@ impl<'a, 'b> plugin::Plugin<Request<'a, 'b>> for Raw {
 
         if need_read {
             let max_length = req.get::<persistent::Read<MaxBodyLength>>()
-                .ok().cloned().unwrap_or(DEFAULT_BODY_LIMIT);
-            let body = try!(read_body_as_utf8(req, max_length));
+                .ok().unwrap_or(std::sync::Arc::new(DEFAULT_BODY_LIMIT));
```

Comment:
Oh, `cloned` is now only available for `Option<&T>`, it used to be for all `D: Deref<Target=T>`. Just do `.map(|x| *x)` instead since `usize` is `Copy`.